web/src/stream_settings_ui.js

```diff
@@ -241,15 +239,6 @@ export function add_sub_to_table(sub) {
         // ID isn't known yet.  These are appended to the top of the
         // list, so they are more visible.
         stream_ui_updates.row_for_stream_id(sub.stream_id).trigger("click");
-        const context = {
-            banner_type: compose_banner.SUCCESS,
-            classname: "stream_creation_confirmation",
-            stream_name: sub.name,
-            stream_url: hash_util.by_stream_url(sub.stream_id),
-        };
-        $("#stream_settings .stream-creation-confirmation-banner").html(
-            render_stream_creation_confirmation_banner(context),
-        );
```

Comment:
So I think we actually don't want to drop this banner, for the following situation:

If you're an administrator creating a channel that you' re not going to be subscribed to, then with this new code, you will not be redirected to the channel. This is probably good -- that's likely a scenario where you're making one or more channels for other people to use.

But in that case, the confirmation banner that you made the channel successfully is important.

Reply:
I pulled this commit off, which helps, but I see two bugs:
- We still show the "We navigated you" modal even when we didn't because we're not a subscriber; I think that code needs to move too.
- It seems like the URL changes but the navigation doesn't actually occur (I'm left still in channel settings with a URL hash that is not a channels settings hash) when I tried to test this end-to-end including myself as a subscriber. Probably this is somehow my fault from fiddling with the PR.

Can you do a careful test and fix at least these two corner cases?

Reply:
> We still show the "We navigated you" modal even when we didn't because we're not a subscriber; I think that code needs to move too.

Done.

> It seems like the URL changes but the navigation doesn't actually occur (I'm left still in channel settings with a URL hash that is not a channels settings hash) when I tried to test this end-to-end including myself as a subscriber. Probably this is somehow my fault from fiddling with the PR.

Yeah, there was a small one-liner bug in the code you pushed. That resulted in NOT navigating.